### PR TITLE
Fix/date time picker to controlled

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
DateTimePicker - turn to a controlled component. Fix issue #729 

## Changelog
DateTimePicker - turn to a controlled component.

### CODE REVIEW

File: src/components/textField/index.js [hunk 1]
```diff
@@ -197,7 +197,8 @@ export default class TextField extends BaseInput {
       value: props.value, // for floatingPlaceholder functionality
```
Comment (ethanshar) [inline src/components/textField/index.js:197]: You created the branch again from another branch. 
This change/file should be in this PR

File: src/helpers/Constants.ts [hunk 2]
```diff
@@ -136,7 +136,10 @@ const constants = {
   /* Accessibility */
```
Comment (ethanshar) [inline src/helpers/Constants.ts:136]: Also this file, I think it belong to another PR

File: src/components/dateTimePicker/index.js [hunk 3]
```diff
@@ -90,17 +92,23 @@ class DateTimePicker extends BaseComponent {
...
-    this.styles = createStyles(this.props);
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.value !== prevState.prevValue) {
+      return {
+        prevValue: prevState.value,
+        value: nextProps.value
+      };
+    }
+    return null;
   }
```
Comment (ethanshar) [inline src/components/dateTimePicker/index.js:112]: There's no need to keep the value in state if the component is controlled.

File: src/components/dateTimePicker/index.js [hunk 4]
```diff
@@ -133,14 +141,14 @@ class DateTimePicker extends BaseComponent {
...
         this.chosenDate = new Date();
       }
-      
+
       _.invoke(this.props, 'onChange', this.chosenDate);
       this.setState({value: this.chosenDate});
     });
 
   getStringValue = () => {
     const {value} = this.state;
```
Comment (ethanshar) [inline src/components/dateTimePicker/index.js:150]: If the component is controlled, the value should derived directly from props and not the state.

**Review by ethanshar — CHANGES_REQUESTED** [review]
Few comments on this PR 
- The PR doesn't change the component to be a controlled component cause it still keep the value in an internal state instead of getting it from the user props. 
- This is a breaking change to the component and we can't push it directly to master.
It means the user must pass the value prop, otherwise it won't react to changes.
It should be part of v6. 
- We need to make sure One App users migrate accordingly before we merge this.

File: src/components/dateTimePicker/index.js (hunk 4 above)
Comment (Inbal-Tish) [reply src/components/dateTimePicker/index.js:150]: You're right, generally speaking, but when we want a controlled component and still need to keep an internal state that's the way to do it. See Carousel, DialogDismissibleView, PageControl, Picker, GradientSlider for example.

File: src/components/dateTimePicker/index.js (hunk 3 above)
Comment (Inbal-Tish) [reply src/components/dateTimePicker/index.js:112]: I need to keep an internal state for the onChange

**Comment by Inbal-Tish** [comment]
@ethanshar I fixed the branching, should be only 1 file change. I don't see a breaking change here, the user doesn't have to pass 'value' prop. The change is made for when the user does pass 'value' prop and change it with screen state change.

File: src/components/textField/index.js (hunk 1 above)
Comment (Inbal-Tish) [reply src/components/textField/index.js]: Done.

**Review by ethanshar — APPROVED** [review]
